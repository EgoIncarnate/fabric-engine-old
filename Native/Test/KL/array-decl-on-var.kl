/*
 *  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
 */

operator entry()
{
  var Integer foo[], bar[4][];
  foo.push( 18 );
  report foo;
  bar[2].push( 42 );
  report bar;
}
