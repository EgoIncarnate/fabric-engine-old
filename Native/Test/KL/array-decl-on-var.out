GlobalList
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 'foo', Integer[] )
          VarDecl( 'bar', Integer[][4] )
        ExprStatement
          MethodOp( 'push' )
            Var('foo')
            ArgList
              ConstSize( 18 )
        Report
          Var('foo')
        ExprStatement
          MethodOp( 'push' )
            IndexOp
              Var('bar')
              ConstSize( 2 )
            ArgList
              ConstSize( 42 )
        Report
          Var('bar')
[18]
[[],[],[42],[]]
